lib/renuo/bin-check/master_thread.rb

```diff
@@ -21,8 +22,20 @@ def finalize
       until waiter.empty?
         thread = waiter.next_wait
         @results << thread[:result]
-        exit 1 if @results.last.exit_code == 1
+        exit_with_error if @results.last.exit_code == 1
```

Comment:
I thought you only wanted it to show the failed one? and I also that i should wait till the end and print all of them then? I thought we said it like that when we were talking about the ipa task, but I can also change it :)